File: SAMPLES/CLOJURE/hello.clj
; hello.clj - println, def, str, variables globales

; --- Bonjour le monde ---
(println "Hello, World!")

; --- Variables globales avec def ---
(def nom "Clojure")
(def version 86)

; --- Affichage de variables ---
(println nom)
(println version)

; --- Concatenation avec str ---
(println (str "Bienvenue dans " nom " version " version))
(println (str "1 + 2 = " (+ 1 2)))

; --- Booleens et nil ---
(println true)
(println false)
(println nil)

; --- Caracteres ---
(println \A)
(println \space)

; --- Plusieurs def ---
(def x 42)
(def y 100)
(println (str "x=" x " y=" y " somme=" (+ x y)))

File: SAMPLES/LUA/hello.lua
-- hello.lua : print, variables locales/globales, concatenation

-- Print basique
print("Hello, world!")

-- Variables globales
greeting = "Bonjour"
name = "Lua"

-- Concatenation de chaines
print(greeting .. " " .. name .. "!")

-- Variables locales
local x = 42
local y = 10
print(x)
print(y)

-- Expressions arithmetiques simples
local sum = x + y
print(sum)

-- Affectation multiple
local a, b, c = 1, 2, 3
print(a)
print(b)
print(c)

-- Variable locale sans init (nil)
local d
print(d)

-- Reassignation
x = x + 1
print(x)

-- Concatenation avec tostring
local msg = "La reponse est " .. tostring(42)
print(msg)

-- tonumber
local n = tonumber("123")
print(n)

-- type
print(type(42))
print(type("hello"))
print(type(nil))
print(type(true))
